adminSiteServer/multiDim.ts

```diff
@@ -329,3 +324,57 @@ export async function createMultiDimConfig(
     }
     return multiDimId
 }
+
+async function getChartConfigsByIds(
+    knex: db.KnexReadonlyTransaction,
+    ids: string[]
+) {
+    const rows = await knex<DbRawChartConfig>(ChartConfigsTableName)
+        .select("id", "patch", "full")
+        .whereIn("id", ids)
+    return new Map(rows.map((row) => [row.id, parseChartConfigsRow(row)]))
+}
+
+export async function setMultiDimPublished(
+    knex: db.KnexReadWriteTransaction,
+    multiDim: DbEnrichedMultiDimDataPage,
+    published: boolean
+) {
+    const chartConfigs = await getChartConfigsByIds(
+        knex,
+        multiDim.config.views.map((view) => view.fullConfigId)
+    )
+
+    await Promise.all(
```

Comment:
I'm unsure about the limitless concurrency this can run in, if the mdim has many views.
Maybe using `p-map` with `{concurrency: 10}` or something is better suited here?

Reply:
I'm happy to change it, and I'd be curious to hear if this was an issue in practice before.

Reply:
I'm not quite sure honestly!
I think undici (Node's fetch implementation) limits the number of open connections by itself, and I believe we have run into some kind of error related to that in the past?

Reply:
Yeah, go for it! It's all just vibes-based anyway 😁

Reply:
Hmm, in that case, I'd probably leave it as `Promise.all()` and fix it if we ever hit a problem. The S3 client doesn't seem to use undici.